core/src/main/java/io/questdb/cairo/GeoHashes.java

```diff
@@ -79,17 +79,26 @@ public static long bitmask(int count, int shift) {
         return ((1L << count) - 1) << shift;
     }
 
-    public static long fromBitString(CharSequence bits) throws NumericException {
-        if (bits.length() > MAX_BITS_LENGTH) {
-            throw NumericException.INSTANCE;
+    public static long fromBitStringNl(CharSequence bits, int start) throws NumericException {
+        int len = bits.length();
+        if (len - start <= 0) {
+            return NULL;
         }
+        return fromBitString(bits, start, Math.min(bits.length(), MAX_BITS_LENGTH));
+    }
+
+    public static long fromBitString(CharSequence bits, int start) throws NumericException {
+        return fromBitString(bits, start, Math.min(bits.length(), MAX_BITS_LENGTH));
+    }
+
+    private static long fromBitString(CharSequence bits, int start, int len) throws NumericException {
         long result = 0;
-        for (int i = 0, n = bits.length(); i < n; i++) {
+        for (int i = start; i < len; i++) {
```

Comment:
This looks like a bug. `len` is not `lenthg`, it's used as `high` 
the overloads should call this methods as 
`fromBitString(bits, start, start + Math.min(bits.length(), MAX_BITS_LENGTH));`

notice the last parameter is `start + ...`

Please add tests with start > 0

Reply:
will rename to limit

Reply:
there are already tests for that, the function parser uses this method and has tests

Reply:
Calls should be 

`fromBitString(bits, start, Math.min(bits.length(), start + MAX_BITS_LENGTH))`

This is the test for this corner case

```
    @Test
    public void testGeoLiteralBinLength() throws Exception {
        assertMemoryLeak(() -> {
            CharSink bitString = new StringSink();
            bitString.put(Chars.repeat("0", 59)).put('1');
            assertSql("select ##" + bitString + " as geobits", "geobits\n" +
                    "000000000001\n");
        });
    }
```

Reply:
wow, thank you for this, `fromBitString(bits, start, Math.min(bits.length(), start + MAX_BITS_LENGTH));` is the correct version